dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/ContentStorageHelper.java

```diff
@@ -0,0 +1,183 @@
+package com.dotcms.rest.api.v1.storage;
+
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.storage.ContentletMetadata;
+import com.dotcms.storage.ContentletMetadataAPI;
+import com.dotcms.storage.FileStorageAPI;
+import com.dotcms.storage.StoragePersistenceAPI;
+import com.dotcms.storage.StoragePersistenceProvider;
+import com.dotcms.storage.StoragePersistenceProvider.INSTANCE;
+import com.dotcms.storage.StorageType;
+import com.dotcms.uuid.shorty.ShortyId;
+import com.dotcms.uuid.shorty.ShortyIdAPI;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import javax.ws.rs.BadRequestException;
+import org.apache.commons.lang.RandomStringUtils;
+
+/**
+ * Helper for the {@link ContentStorageResource}
+ * @author jsanca
+ */
+class ContentStorageHelper {
+
+    private final ContentletAPI contentletAPI;
+    private final ShortyIdAPI shortyIdAPI;
+    private final FileStorageAPI fileStorageAPI;
```

Comment:
Codacy found an issue: [Perhaps 'fileStorageAPI' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)